src/PantherTestCaseTrait.php

```diff
@@ -69,61 +71,69 @@
-            self::$webServerManager = null;
+        foreach (self::$webServerManagers as $webServerManager) {
+            $webServerManager->quit();
         }
 
-        if (null !== self::$pantherClient) {
-            self::$pantherClient->quit();
-            self::$pantherClient = null;
-        }
+        self::$webServerManagers = [];
 
-        if (null !== self::$goutteClient) {
-            self::$goutteClient = null;
+        foreach (self::$pantherClients as $pantherClient) {
+            $pantherClient->quit();
         }
 
+        self::$pantherClients = [];
+
+        self::$goutteClients = [];
+
         self::$baseUri = null;
     }
 
     /**
      * @param array $options see {@see $defaultOptions}
      */
-    public static function startWebServer(array $options = []): void
+    public static function startWebServer(array $options = []): string
     {
-        if (null !== static::$webServerManager) {
-            return;
-        }
-
         if ($externalBaseUri = $options['external_base_uri'] ?? $_SERVER['PANTHER_EXTERNAL_BASE_URI'] ?? self::$defaultOptions['external_base_uri']) {
             self::$baseUri = $externalBaseUri;
 
-            return;
+            return $externalBaseUri;
         }
 
-        $options = [
-            'webServerDir' => $options['webServerDir'] ?? static::$webServerDir ?? $_SERVER['PANTHER_WEB_SERVER_DIR'] ?? self::$defaultOptions['webServerDir'],
-            'hostname' => $options['hostname'] ?? self::$defaultOptions['hostname'],
-            'port' => (int) ($options['port'] ?? $_SERVER['PANTHER_WEB_SERVER_PORT'] ?? self::$defaultOptions['port']),
-            'router' => $options['router'] ?? $_SERVER['PANTHER_WEB_SERVER_ROUTER'] ?? self::$defaultOptions['router'],
-        ];
+        $hostname = $options['hostname'] ?? self::$defaultOptions['hostname'];
+        $port = (int) ($options['port'] ?? $_SERVER['PANTHER_WEB_SERVER_PORT'] ?? self::$defaultOptions['port']);
+
+        $baseUri = sprintf('http://%s:%s', $hostname, $port);
+        if (!isset(self::$webServerManagers[$baseUri])) {
+            $webServerDir = $options['webServerDir'] ?? static::$webServerDir ?? $_SERVER['PANTHER_WEB_SERVER_DIR'] ?? self::$defaultOptions['webServerDir'];
+            $router = $options['router'] ?? $_SERVER['PANTHER_WEB_SERVER_ROUTER'] ?? self::$defaultOptions['router'];
+
+            $webServerManager = new WebServerManager($webServerDir, $hostname, $port, $router);
+            $webServerManager->start();
 
-        self::$webServerManager = new WebServerManager(...array_values($options));
-        self::$webServerManager->start();
+            self::$baseUri = $baseUri;
 
-        self::$baseUri = sprintf('http://%s:%s', $options['hostname'], $options['port']);
+            self::$webServerManagers[$baseUri] = $webServerManager;
+        }
+
+        return $baseUri;
     }
 
-    public static function isWebServerStarted()
+    public static function isWebServerStarted(): bool
```

Comment:
This might also be a problem. Maybe it would need an argument to know if a particular web server was started (so a need to name them ?)